File: Programmers/64061/Solution.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;

        int bHeigth = board.length;
        int mLen = moves.length;

        Stack<Integer> items = new Stack();

        for (int i = 0; i < mLen; i++) {
            int index = moves[i] - 1;
            int item = 0;
            for (int j = 0; j < bHeigth; j++) {
                if (board[j][index] != 0) {
                    item = board[j][index];
                    board[j][index] = 0;
                    break;
                }
            }
            if (!items.isEmpty() && items.peek() == item) {
                items.pop();
                answer += 2;
            } else if (item != 0) {
                items.push(item);
            }
        }

        return answer;
    }
}

File: Programmers/64061/solution.py
def solution(board, moves):
    answer = 0

    depth = len(board)
    result = []

    for move in moves:
        for d in range(depth):
            # 인형을 꺼냄
            toy = board[d][move - 1]
            # 인형이 있을 경우
            if toy != 0:
                # 인형을 꺼내서 0으로 변경
                board[d][move - 1] = 0
                if len(result) > 0:
                    if result[-1] == toy:
                        result.pop()
                        answer += 1
                        break
                    else:
                        result.append(toy)
                        break
                else:
                    result.append(toy)
                    break

    return answer * 2
